api/client.go

```diff
@@ -329,13 +331,52 @@ func (c *Client) AddLocalCharm(curl *charm.URL, ch charm.Charm) (*charm.URL, err
 		return nil, errors.Errorf("unknown charm type %T", ch)
 	}
 
-	curl, err := c.UploadCharm(curl, archive)
+	anyHooks, err := hasHooks(archive.Name())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if !anyHooks {
+		return nil, errors.Errorf("invalid charm %q: has no hooks", curl.Name)
+	}
+
+	curl, err = c.UploadCharm(curl, archive)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	return curl, nil
 }
 
+var hasHooks = hasHooksFolder
+
+func hasHooksFolder(name string) (bool, error) {
```

Comment:
I guess this is covered by feature test.
Would be good to see unit test as well.

Reply:
added